test/chat/db/scope/cargo_key_scope_test.exs

```diff
@@ -1,41 +1,129 @@
 defmodule Chat.Db.Scope.CargoKeyScopeTest do
   use ExUnit.Case, async: false
 
-  alias Chat.Db.Scope.KeyScope
-  alias Chat.Dialogs
-  alias Chat.Identity
-  alias Chat.Messages
-  alias Chat.RoomInviteIndex
-  alias Chat.Rooms
-  alias Chat.User
-
-  test "room and invitations in dialogs should be selected" do
+  alias Chat.{
+    AdminRoom,
+    Db.Scope.KeyScope,
+    Dialogs,
+    Identity,
+    Messages,
+    RoomInviteIndex,
+    Rooms,
+    User
+  }
+
+  setup do
     {bot_key, room_key} = generate_user_with_dialogs_content_and_invite_in_room()
+    {:ok, %{bot_key: bot_key, room_key: room_key}}
+  end
+
+  test "room and invitations in dialogs should be selected", %{
+    bot_key: bot_key,
+    room_key: room_key
+  } do
+    assert_keys_for_cargo_keys(room_key, [bot_key], 1)
+  end
+
+  test "room invitation should be found in dialog of operator and user by only checkpoints key" do
+    {admin, john, operator} = setup_test_data()
+    [checkpoint1_key, checkpoint2_key] = generate_checkpoints([admin, john])
 
+    room_identity = generate_cargo_room(operator)
+    room_key = Identity.pub_key(room_identity)
+
+    generate_dialogs_and_cargo_room_invite([admin, john], operator, room_identity)
+    user_c = User.login("UserC") |> create_user_from_identity()
+    generate_dialogs_and_cargo_room_invite([user_c], operator, room_identity)
+
+    user_d = User.login("UserD") |> create_user_from_identity()
+    generate_dialogs_and_cargo_room_invite([user_d], user_c, room_identity)
+
+    assert_keys_for_cargo_keys(room_key, [checkpoint1_key, checkpoint2_key], 4)
+  end
+
```

Comment:
add a test with 
- checkpoints c1,c2,c3
- operarators o1, o2
- users u1,u2,u3, u4

room was created by o1 with checkpoints list c1 and c2
later o2 added manual invite for c3 into room
o1 invited u1 
u1 invited u2, u4
u2 invited u3
u3 invited u4 and u1

u3's invite should not get into scope

Reply:
test rewrited with dynamic setup and described case